### PR TITLE
check_dns: Add --expect-nxdomain

### PR DESCRIPTION
**Pull request by Jonny007-MKD**
I want to check that certain domains are blocked and return a NXDOMAIN (that's how my adblock script configures my unbound instance). I added this option to check_dns.

### CODE REVIEW

File: plugins/t/check_dns.t [hunk 1]
```diff
@@ -58,7 +58,7 @@ my $dns_server       = getTestParameter(
 my $host_nonresponsive = getTestParameter(
 			"NP_HOST_NONRESPONSIVE",
 			"The hostname of system not responsive to network requests",
-			"10.0.0.1",
+			"192.0.2.0",
```
Comment (tobiaswiese) [inline plugins/t/check_dns.t:61]: is this change really intended here?

Comment (Jonny007-MKD) [reply plugins/t/check_dns.t:61]: No. I can't recall why I changed that...

**Comment by RincewindsHat** [comment]
Just tried it.  Works for me. Would merge.

**Comment by Jonny007-MKD** [comment]
ping @sni

**Comment by sni** [comment]
looks good from a quick look, could you please rebase this onto the latest master so our unit tests would run?

**Review by sni — APPROVED** [review]

**Comment by Jonny007-MKD** [comment]
ok, I used the Fetch & Merge feature from Github. Seems like it made a merge commit and didn't rebase (which I hoped for). Tests failed with
> E: Package 'netcat' has no installation candidate

**Comment by sni** [comment]
please try to rebase or merge again. Tests should be fine now.

**Comment by sni** [comment]
thanks, merged it.
